stdlib/public/core/Stride.swift

```diff
@@ -171,6 +224,39 @@ extension Strideable {
   }
 }
 
+extension Strideable where Self: FixedWidthInteger & SignedInteger {
+  @_alwaysEmitIntoClient
+  public static func _step(
+    after current: (index: Int?, value: Self),
+    from start: Self, by distance: Self.Stride
+  ) -> (index: Int?, value: Self) {
+    let value = current.value
+    let (partialValue, overflow) =
+      Self.bitWidth >= Self.Stride.bitWidth ||
+        (value < (0 as Self)) == (distance < (0 as Self.Stride))
+          ? value.addingReportingOverflow(Self(distance))
+          : (Self(Self.Stride(value) + distance), false)
+    return overflow
+      ? (.min, distance < (0 as Self.Stride) ? .min : .max)
+      : (.max, partialValue)
```

Comment:
Nit: we should return `nil` for the index in the regular case -- `.max` reads like it might have a hidden meaning like `.min` does.

Reply:
Sure, we're no longer using `.max` for any particular purpose, so this can come out.